From d4ca9d3c3f6a6d1a76ca4581b4d33e6a9c94574b Mon Sep 17 00:00:00 2001
From: zhuangqiubin <zhuangqiubin@allwinnertech.com>
Date: Sun, 27 Sep 2020 16:57:56 +0800
Subject: [PATCH] use cmake 3.4

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fd67ae0..fbe43bb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,7 +17,7 @@
 
 
 project(librsync C)
-cmake_minimum_required(VERSION 3.6)
+cmake_minimum_required(VERSION 3.4)
 
 INCLUDE(CMakeDependentOption)
 include(GNUInstallDirs)
-- 
2.28.0

