--- a/tools/Makefile.in	2020-02-19 13:33:44.092097657 +0800
+++ b/tools/Makefile.in	2020-02-19 13:34:52.832092916 +0800
@@ -124,7 +124,7 @@

 dmsetup: dmsetup.o $(top_builddir)/libdm/libdevmapper.$(LIB_SUFFIX)
	$(CC) $(CFLAGS) $(LDFLAGS) $(EXTRA_EXEC_LDFLAGS) $(ELDFLAGS) \
-	      -o $@ dmsetup.o -ldevmapper $(LIBS)
+	      -o $@ dmsetup.o -ldevmapper $(M_LIBS) $(PTHREAD_LIBS) $(LIBS)

 dmsetup.static: dmsetup.o $(interfacebuilddir)/libdevmapper.a
	$(CC) $(CFLAGS) $(LDFLAGS) -static -L$(interfacebuilddir) \
@@ -138,7 +138,7 @@

 lvm: $(OBJECTS) lvm.o $(top_builddir)/lib/liblvm-internal.a
	$(CC) $(CFLAGS) $(LDFLAGS) $(EXTRA_EXEC_LDFLAGS) $(ELDFLAGS) -o $@ $(OBJECTS) lvm.o \
-		$(LVMLIBS) $(READLINE_LIBS) $(LIBS)
+		$(LVMLIBS) $(READLINE_LIBS) $(M_LIBS) $(PTHREAD_LIBS) $(LIBS)

 DEFS_man-generator.o += -DMAN_PAGE_GENERATOR
