--- /source/backend/opencl/execution/cl/reduction.cl	2020-05-07 18:22:11.000000000 +0800
+++ /source/backend/opencl/execution/cl/reduction.cl	2020-09-27 17:23:33.261240420 +0800
@@ -245,7 +245,7 @@
 __kernel void reduct_2d(GLOBAL_SIZE_3_DIMS
                         __read_only image2d_t input,
                         __write_only image2d_t output,
-                        __global FLOAT4 *groupBuffer,
+                        __local FLOAT4 *groupBuffer,
                         __global FLOAT *leftBuffer,
                         __private const int groupWorkSize,
                         __private const int computeNum,
