--- a/src/utils/common.c
+++ b/src/utils/common.c
@@ -462,11 +462,85 @@ TCHAR * wpa_strdup_tchar(const char *str
 }
 #endif /* CONFIG_NATIVE_WINDOWS */
 
+//add by cx2668
+static unsigned char char_to_hex(unsigned char x)
+{
+	return (unsigned char)(x > 9 ? x + 55 : x + 48);
+}
+
+static int is_alpha_number_char(unsigned char c)
+{
+	if ((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9'))
+		return 1;
+	return 0;
+}
+
+//url编码实现
+void urlencode(unsigned char *src, int src_len, unsigned char *dest, int dest_len)
+{
+	unsigned char ch;
+	int len = 0;
+
+	while (len < (dest_len - 4) && *src)
+	{
+		ch = (unsigned char)*src;
+		if (*src == ' ')
+		{
+			*dest++ = '+';
+		}
+		else if (is_alpha_number_char(ch) || strchr("-_.!~*'()", ch))
+		{
+			*dest++ = *src;
+		}
+		else
+		{
+			*dest++ = '%';
+			*dest++ = char_to_hex((unsigned char)(ch >> 4));
+			*dest++ = char_to_hex((unsigned char)(ch % 16));
+		}
+		++src;
+		++len;
+	}
+	*dest = 0;
+	return;
+}
+
+//解url编码实现
+unsigned char *urldecode(unsigned char *encd, unsigned char *decd)
+{
+	int j, i;
+	char *cd = encd;
+	char p[2];
+	unsigned int num;
+	j = 0;
+
+	for (i = 0; i < strlen(cd); i++)
+	{
+		memset(p, '\0', 2);
+		if (cd[i] != '%')
+		{
+			decd[j++] = cd[i];
+			continue;
+		}
+
+		p[0] = cd[++i];
+		p[1] = cd[++i];
+
+		p[0] = p[0] - 48 - ((p[0] >= 'A') ? 7 : 0) - ((p[0] >= 'a') ? 32 : 0);
+		p[1] = p[1] - 48 - ((p[1] >= 'A') ? 7 : 0) - ((p[1] >= 'a') ? 32 : 0);
+		decd[j++] = (unsigned char)(p[0] * 16 + p[1]);
+	}
+	decd[j] = '\0';
+
+	return decd;
+}
 
 void printf_encode(char *txt, size_t maxlen, const u8 *data, size_t len)
 {
 	char *end = txt + maxlen;
-	size_t i;
+	size_t i , j;
+	char *head = txt;
+	int URLDecodeFlag = 0;
 
 	for (i = 0; i < len; i++) {
 		if (txt + 4 >= end)
@@ -509,6 +583,38 @@ void printf_encode(char *txt, size_t max
 	}
 
 	*txt = '\0';
+
+	//add by cx2668, for Chinese SSID
+	char *tmpSSIDTxt = (char *)os_malloc(sizeof(char) * (strlen(head) + 1));
+	if(tmpSSIDTxt == NULL)
+	{
+		printf("tmpSSIDTxt malloc error.\n");
+		return;
+	}
+	os_memset(tmpSSIDTxt , 0 , strlen(head) + 1);
+	for(i = 0, j = 0; i <= strlen(head);)
+	{
+		if(head[i] == '\\' && head[i + 1] == 'x')
+		{
+			URLDecodeFlag = 1;
+			tmpSSIDTxt[j] = '%';
+			i += 2;
+		}
+		else
+		{
+			tmpSSIDTxt[j] = head[i];
+			i++;
+		}
+		j++;
+	}
+
+	if(URLDecodeFlag)
+    {
+        os_memset(head , 0 , strlen(tmpSSIDTxt) + 1);
+		urldecode(tmpSSIDTxt , head);
+    }
+
+	os_free(tmpSSIDTxt);
 }
 
 
