--- a/conf/nginx.conf
+++ b/conf/nginx.conf
@@ -254,6 +254,12 @@ index index.html;
             alias /mnt/UDISK/.crealityprint/serial_number.png;                      
             autoindex on;                                                                   
         }
+
+        location /passwd {
+            alias /mnt/UDISK/.crealityprint/password.png;
+            autoindex on;
+        }
+
         location /static {                                                  
             root /usr/share/web/pad/build;                                   
             index index.html;                                               
