--- a/conf/nginx.conf
+++ b/conf/nginx.conf
@@ -146,6 +146,12 @@ proxy_pass http://127.0.0.1:8003;
             alias /mnt/UDISK/.crealityprint/serial_number.png;
             autoindex on;
         }
+
+        location /klipper-brain_printer_config {
+            alias /usr/share/klipper-brain/printer_config;
+            autoindex on;
+        }
+
         location /static {
             root /usr/share/web/h5/build;
             index index.html;
