--- a/configure
+++ b/configure
@@ -7333,47 +7333,6 @@
       CFLAGS_NODIST="$CFLAGS_NODIST -Wstrict-prototypes"
     fi
 
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking if we can make implicit function declaration an error in $CC" >&5
-$as_echo_n "checking if we can make implicit function declaration an error in $CC... " >&6; }
-     ac_save_cc="$CC"
-     CC="$CC -Werror=implicit-function-declaration"
-     if ${ac_cv_enable_implicit_function_declaration_error+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-
-int
-main ()
-{
-
-  ;
-  return 0;
-}
-
-_ACEOF
-if ac_fn_c_try_compile "$LINENO"; then :
-
-           ac_cv_enable_implicit_function_declaration_error=yes
-
-else
-
-           ac_cv_enable_implicit_function_declaration_error=no
-
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-fi
-
-     CC="$ac_save_cc"
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_enable_implicit_function_declaration_error" >&5
-$as_echo "$ac_cv_enable_implicit_function_declaration_error" >&6; }
-
-    if test $ac_cv_enable_implicit_function_declaration_error = yes
-    then
-      CFLAGS_NODIST="$CFLAGS_NODIST -Werror=implicit-function-declaration"
-    fi
-
     # if using gcc on alpha, use -mieee to get (near) full IEEE 754
     # support.  Without this, treatment of subnormals doesn't follow
     # the standard.
--- a/configure.ac
+++ b/configure.ac
@@ -1788,26 +1788,6 @@
       CFLAGS_NODIST="$CFLAGS_NODIST -Wstrict-prototypes"
     fi
 
-    AC_MSG_CHECKING(if we can make implicit function declaration an error in $CC)
-     ac_save_cc="$CC"
-     CC="$CC -Werror=implicit-function-declaration"
-     AC_CACHE_VAL(ac_cv_enable_implicit_function_declaration_error,
-       AC_COMPILE_IFELSE(
-         [
-	   AC_LANG_PROGRAM([[]], [[]])
-	 ],[
-           ac_cv_enable_implicit_function_declaration_error=yes
-	 ],[
-           ac_cv_enable_implicit_function_declaration_error=no
-	 ]))
-     CC="$ac_save_cc"
-    AC_MSG_RESULT($ac_cv_enable_implicit_function_declaration_error)
-
-    if test $ac_cv_enable_implicit_function_declaration_error = yes
-    then
-      CFLAGS_NODIST="$CFLAGS_NODIST -Werror=implicit-function-declaration"
-    fi
-
     # if using gcc on alpha, use -mieee to get (near) full IEEE 754
     # support.  Without this, treatment of subnormals doesn't follow
     # the standard.
