--- a/Python/pytime.c
+++ b/Python/pytime.c
@@ -709,6 +709,7 @@ pygettimeofday(_PyTime_t *tp, _Py_clock_
         info->monotonic = 0;
         info->adjustable = 1;
         if (clock_getres(CLOCK_REALTIME, &res) == 0) {
+            res.tv_nsec = 1;
             info->resolution = res.tv_sec + res.tv_nsec * 1e-9;
         }
         else {
@@ -906,6 +907,7 @@ pymonotonic(_PyTime_t *tp, _Py_clock_inf
             PyErr_SetFromErrno(PyExc_OSError);
             return -1;
         }
+        res.tv_nsec = 1;
         info->resolution = res.tv_sec + res.tv_nsec * 1e-9;
     }
     if (pytime_fromtimespec(tp, &ts, raise) < 0) {
